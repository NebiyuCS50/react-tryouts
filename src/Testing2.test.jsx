import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Test2 from "./Test2.jsx";

describe("Test2 Component", () => {
  it("renders magnificent monkeys", () => {
    const { container } = render(<Test2 />);
    expect(container.textContent).toMatchSnapshot();
  });
  it("renders radical rhinos after button click", async () => {
    const user = userEvent.setup();
    render(<Test2 />);
    const button = screen.getByRole("button", { name: /click me/i });
    await user.click(button);
    expect(screen.getByRole("heading").textContent).toMatch(/Radical Rhinos/i);
  });
});
