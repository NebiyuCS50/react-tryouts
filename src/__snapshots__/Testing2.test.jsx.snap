// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test2 Component > renders magnificent monkeys 1`] = `"Click MeMagnificent Monkeys"`;
